designs/2019-typescript-eslintrc-support/README.md

```diff
@@ -0,0 +1,351 @@
+- Start Date: 2019-11-18
+- RFC PR: https://github.com/eslint/rfcs/pull/50
+- Authors: Gareth Jones ([@g-rath](https://github.com/g-rath))
+
+# First-Class Support For `.eslintrc.ts`
+
+## Summary
+
+Support the ability to consume `.eslintrc.ts` configuration files natively.
+
+## Motivation
+
+Currently configuration settings are validated by JSON Schema, which while
+powerful does not have ubiquitous & high quality support across editors &
+tooling.
+
+It also tends to generate somewhat cryptic messages, especially when using
```

Comment:
Although typescript made some improvements in that regard, the error messages from tsc can as cryptic as with JSON Schema (IMHO).

Reply:
The difference being IMO that typescript is constantly working to improve the error messages to make them less cryptic. Every version they ship improvements to do things like better refine the actual source of the error, or reduce useless error message nesting.

In comparison to JSON Schema, which has a fixed set of messages, and relies upon the implementor (in this case eslint) to catch the error message and apply their own clarifications.